features/api-layer/src/main/java/org/opennms/features/apilayer/collectors/ServiceCollectorImpl.java

```diff
@@ -117,6 +133,12 @@ public String getEffectiveLocation(String location) {
         return serviceCollectorFactory.unmarshalParameters(parameters);
     }
 
+    @Override
+    public String getCollectorClassName() {
+        return serviceCollectorFactory.getCollectorClassName();
```

Comment:
I'm not sure if I get this correctly: The class name here is used to instantiate a `org.opennms.netmgt.collection.api.ServiceCollector` implementation in the Collector RPC Client Module. But it looks like this will return a class name of a `org.opennms.integration.api.v1.collectors.ServiceCollector` implementation.

If this is true, a possible fix would be to just transport the delegated class name of the `ServiceCollectorFactory` as a parameter to a new collector implementation, which in turn could instantiate the factory on the minion and call the created collector in a appropriate way.

Reply:
Agreed, I think we can find a better way to achieve this.